### PR TITLE
ZIOS-10335: Attachments are not deleted when removed individually

### PR DESCRIPTION
**Pull request by nicorsm**
### Issues

A copy of an unencrypted downloaded file that resides on the `/tmp` folder of the file system is not removed when deleting the message from a conversation.

### Causes

The file was not deleted. The only copy of the (encrypted?) file that got deleted was the one inside the `zm_fileAssetCache` folder (which is in `/Library/Caches`).

### Solutions

I've added some logic (and a test) to delete both copies of the file.

### CODE REVIEW

File: Source/Model/Message/ZMAssetClientMessage+Deletion.swift [hunk 1]
```diff
@@ -29,6 +29,12 @@ extension ZMAssetClientMessage {
 
     func deleteContent() {
         self.managedObjectContext?.zm_fileAssetCache.deleteAssetData(self)
+        
+        if let url = temporaryDirectoryURL,
+            FileManager.default.fileExists(atPath: url.path) {
+            try? FileManager.default.removeItem(at: url)
```
Comment (mikeger) [inline Source/Model/Message/ZMAssetClientMessage+Deletion.swift]: Should it delete all the assets in that directory when one asset is deleted?

Comment (nicorsm) [reply Source/Model/Message/ZMAssetClientMessage+Deletion.swift]: Yes, actually we're creating a folder for each file, based on the `cacheKey`, as you can see here:
screen shot 2018-08-20 at 15 51 00

I thought it was good to clear the folder in order to keep the parent folder clean. Do you know any case where we use those folders for 2+ files?

Comment (mikeger) [reply Source/Model/Message/ZMAssetClientMessage+Deletion.swift]: Should be fine, I haven't noticed that `temporaryDirectoryURL` is not the same for all files (looking on the implementation).

**Comment by nicorsm** [comment]
@zenkins release patch

**Pull request merged**